api/repositories/pod_repository.go

```diff
@@ -125,12 +124,7 @@ func (r *PodRepo) ListPodStats(ctx context.Context, authInfo authorization.Info,
 
 		podMetrics, err := r.metricsFetcher(ctx, p.Namespace, p.Name)
 		if err != nil {
-			errorMsg := err.Error()
-			if strings.Contains(errorMsg, "not found") ||
-				strings.Contains(errorMsg, "the server could not find the requested resource") {
-				continue
-			}
-			return nil, err
+			continue
```

Comment:
Not related to this change, but there is no logging of the failure here. As that currently only happens in the api layer and we don't pass an error down to it here, we won't have a chance to log it.

Reply:
There's no reason why we shouldn't log in the repositories right? I think it might be good to pass the logger down so that when someone asks "why are my metrics always missing?" they can find some logs to answer that question.

Reply:
Yep - absolutely.

I think we have to find a good option for passing the loggers through the code. We inject one containing the correct correlation ID in the handlers. It would be good to pass that around, preferably without changing every method in the system to accept a log argument. Piggy-backing on context is an option built into logr.

Reply:
I think we will need to do something to propagate loggers in order to finish #1179. @kieron-dev I will drop a link to your comment into that story as it sounds like it would be worth trying.